x-pack/solutions/security/plugins/cloud_security_posture/server/plugin.test.ts

```diff
@@ -123,6 +123,51 @@ describe('Cloud Security Posture Plugin', () => {
       expect(spy).toHaveBeenCalledTimes(0);
     });
 
+    it('should retry getInstallation on failures', async () => {
+      let callCount = 0;
+      fleetMock.packageService.asInternalUser.getInstallation.mockImplementation(
+        async (): Promise<Installation | undefined> => {
+          callCount++;
+          if (callCount < 3) {
+            throw new Error('ES connection failed');
+          }
+          return {} as jest.Mocked<Installation>;
+        }
+      );
+
+      const context = coreMock.createPluginInitializerContext<unknown>();
+      plugin = new CspPlugin(context);
+      const spy = jest.spyOn(plugin, 'initialize').mockImplementation();
+
+      // Act
+      await plugin.start(coreMock.createStart(), mockPlugins);
+      await mockPlugins.fleet.fleetSetupCompleted();
+
+      // Assert
+      expect(fleetMock.packageService.asInternalUser.getInstallation).toHaveBeenCalledTimes(3);
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should handle getInstallation complete failure after retries', async () => {
+      fleetMock.packageService.asInternalUser.getInstallation.mockImplementation(
+        async (): Promise<Installation | undefined> => {
+          throw new Error('ES connection failed persistently');
```

Comment:
can you also check that this error is propagated up properly so we can catch it outside of retry?